### PR TITLE
docs] - Rename to Dagster+ (DOC-141)

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR updates the docs to use Dagster+ instead of Dagster Cloud:

- Updates copy
- Renames directory and adds redirects for all pages formerly in `/dagster-cloud`
- Updates references to Cloud in `/docs/sphinx/sections` - this doesn't include docstrings, if any

What's **not** in this PR:

- References to file names, ex: `dagster_cloud.yaml`
- References contained in files, ex: `dagsterCloudAgent` in the CloudFormation template
- References to Cloud in REST API headers, ex: `Dagster-Cloud-Api-Token`
- Classes in libraries, ex: `dagster_airflow.DagsterCloudOperator`
- Names of env vars used in Dagster+, ex: `DAGSTER_CLOUD_*`
- References in Dagster+ GitHub Workflow Actions
- Links to any GitHub repo that mentions Cloud, but is *not* `dagster-io/dagster-cloud`
- References in doc code snippets that aren't the `dagster-plus` CLI

## How I Tested These Changes

so much reading

and `yarn test` in `docs/next`

### CODE REVIEW

File: docs/content/dagster-plus/insights.mdx [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 
-# Dagster Cloud Insights (Experimental)
+# Dagster+ Insights (Experimental)
 
 <Note>
   This feature is considered <strong>experimental</strong>.
 </Note>
 
-Using Dagster Cloud Insights, you can gain visibility into historical usage and cost metrics such as Dagster Cloud run duration, credit usage, and failures. You can also [integrate other metrics](#integrating-other-metrics) into the Insights UI and [export metrics from Dagster Cloud](#exporting-metrics).
+Using Dagster+ Insights, you can gain visibility into historical usage and cost metrics such as Dagster+ run duration, credit usage, and failures. You can also [integrate other metrics](#integrating-other-metrics) into the Insights UI and [export metrics from Dagster+](#exporting-metrics).
```
Comment (krukowski) [inline docs/content/dagster-plus/insights.mdx:14]: This is a branding question, but seeing `Dagster+` everywhere feels a little clunky when it's not required to differentiate from OSS (e.g. "Dagster+ run duration")

File: docs/content/dagster-plus/managing-deployments/alerts/email.mdx [hunk 2]
```diff
@@ -90,27 +90,27 @@ Refer to the [Managing alerts using the `dagster-cloud` CLI](/dagster-cloud/mana
...
+    href="/dagster-plus/managing-deployments/alerts"
   ></ArticleListItem>
   <ArticleListItem
-    title="Managing alerts in Dagster Cloud"
-    href="/dagster-cloud/managing-deployments/alerts/managing-alerts-in-ui"
+    title="Managing alerts in Dagster+"
+    href="/dagster-plus/managing-deployments/alerts/managing-alerts-in-ui"
   ></ArticleListItem>
   <ArticleListItem
     title="Managing alerts using the dagster-cloud CLI"
```
Comment (krukowski) [inline docs/content/dagster-plus/managing-deployments/alerts/email.mdx:101]:
```suggestion
    title="Managing alerts using the dagster-plus CLI"
```

File: docs/content/integrations/dbt.mdx [hunk 3]
```diff
@@ -75,7 +74,7 @@ There are a few ways to get started with Dagster and dbt:
 - Play around with a [working dbt + Dagster project](https://github.com/dagster-io/dagster/tree/master/examples/assets_dbt_python).
 - Browse the [dagster-dbt integration reference](/integrations/dbt/reference) for short lessons on Dagster + dbt topics.
```
Comment (krukowski) [inline docs/content/integrations/dbt.mdx:75]: > short lessons on Dagster + dbt

Dagster with dbt? dbt + Dagster?

**Review by krukowski — DISMISSED** [review]
Missing changes to the nav header:
image

Search returns old pages as results when searching for "cloud", then then 404:
image

The subheader under deployments is still called Cloud:
image

Not sure if this is related to these changes or just a vercel preview quirk, but almost all pages give the warning about being outdates, and "View Latest Version" kicks you back out to the dagster.io home page.
image

File: docs/content/dagster-plus/managing-deployments/alerts/email.mdx (hunk 2 above)
Comment (erinkcochran87) [reply docs/content/dagster-plus/managing-deployments/alerts/email.mdx:101]: Good catch, thank you!

File: docs/content/dagster-plus/managing-deployments/alerts/email.mdx [hunk 4]
```diff
@@ -90,27 +90,27 @@ Refer to the [Managing alerts using the `dagster-cloud` CLI](/dagster-cloud/mana
...
-    href="/dagster-cloud/managing-deployments/alerts/managing-alerts-cli"
+    href="/dagster-plus/managing-deployments/alerts/managing-alerts-cli"
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/content/dagster-plus/managing-deployments/alerts/email.mdx:102]: Weird - this didn't happen to me. I'll keep an eye on it

**Comment by erinkcochran87** [comment]
@krukowski Thank you so much for taking a look at this :bear: of a PR. You're doin' the lord's work here.

Since you reviewed this, I:

- Updated the sidenav to `Dagster+` (thanks for catching that!)
- Updated remaining URLs from `dagster.cloud` to `dagster.plus`
- Changed all instances of `<product/Dagster> + <product/Dagster` to use `&` instead - I'm glad you looked at this, because I thought I'd gotten them all last week!

And to address your comments:

- **Nav header**: The design / FE team will be tackling this separately (see FE-292 in Linear)
- **Search results**: Algolia relies on the sitemap in prod to create search results, so until we merge this PR and re-crawl the site, these Cloud results will remain.
- **Outdated docs warning** - Just a Vercel quirk! This will go away in prod.

I'm getting clarification on the style questions you brought up (ex: `Dagster+!`), but I suspect it'll remain as-is even if it looks a bit weird, since it's the proper name of the product. I'll update this if I hear something different, though.

**Review by cmpadden — APPROVED** [review]

**Pull request merged**